addons/l10n_ro_efactura/models/account_move.py

```diff
@@ -0,0 +1,190 @@
+        else:
+            self._l10n_ro_edi_get_sending_and_failed_documents().unlink()
+            self._l10n_ro_edi_create_document_invoice_sending(self, result['key_loading'])
+
+    def _l10n_ro_edi_fetch_invoice_sending_documents(self):
+        """ This method loops over all invoice with sending document in `self`. For each of them,
+            it pre-checks error and make a fetch request for the invoice. Based on the answer, it will then:
+             - if error -> delete all errors, create a new error document
+             - if receive `key_download` -> immediately make a download request and process it:
+                - if error -> delete all sending and error documents, create a new error document
+                - if success -> delete all sending and error documents, create success document
+             - if no answer is received, it will do nothing on the selected invoice
+        """
+        session = requests.session()
+        to_delete_documents = self.env['l10n_ro_edi.document']
+        invoices_to_fetch = self.filtered(lambda inv: inv.l10n_ro_edi_state == 'invoice_sending')
+        self.env['res.company']._with_locked_records(invoices_to_fetch)
+
+        for invoice in invoices_to_fetch:
+            active_sending_document = invoice.l10n_ro_edi_document_ids.filtered(lambda d: d.state == 'invoice_sending')[0]
+
+            if errors := invoice.company_id._l10n_ro_edi_get_errors_pre_request():
+                to_delete_documents |= invoice._l10n_ro_edi_get_failed_documents()
+                self._l10n_ro_edi_create_document_invoice_sending_failed(invoice, '\n'.join(errors))
+                continue
+            result = self.env['l10n_ro_edi.document']._request_ciusro_fetch_status(
+                company=invoice.company_id,
+                key_loading=active_sending_document.key_loading,
+                session=session,
+            )
+
+            if 'error' in result:
+                to_delete_documents |= invoice._l10n_ro_edi_get_sending_and_failed_documents()
+                self._l10n_ro_edi_create_document_invoice_sending_failed(invoice, result['error'])
+            elif 'key_download' in result:
+                # use the obtained key_download to immediately make a download request and process them
+                final_result = self.env['l10n_ro_edi.document']._request_ciusro_download_answer(
+                    company=invoice.company_id,
+                    key_download=result['key_download'],
+                    session=session,
+                )
+                to_delete_documents |= invoice._l10n_ro_edi_get_sending_and_failed_documents()
+                if 'error' in final_result:
+                    self._l10n_ro_edi_create_document_invoice_sending_failed(invoice, final_result['error'])
+                else:
+                    self._l10n_ro_edi_create_document_invoice_sent(invoice, final_result)
```

Comment:
you need to commit between each call
```py
if not modules.module.current_test:
    self.cr.commit()
```
That way, if for some reason the code raises for one invoice, the already processed ones are safe